shell/server/installer.js

```diff
@@ -102,6 +128,28 @@ cancelDownload = function (packageId) {
   }
 }
 
+if (!Meteor.settings.replicaNumber) {
+  installers = {};
+
+  Meteor.startup(function () {
+    // Restart any deletions that were killed while in-progress.
+    Packages.find({status: "delete"}).forEach(function (pkg) {
+      deletePackageInternal(pkg._id);
+    });
+
+    // Watch for new installation requests and fulfill them.
+    Packages.find({status: {$in: ["download", "verify", "unpack", "analyze"]}}).observe({
+      added: startInstallInternal,
+      removed: cancelDownloadInternal
+    });
+
+    // Watch for new cleanup requests and fulfill them.
+    Packages.find({status: "ready", shouldCleanup: true}).observe({
+      added: deletePackageInternal
```

Comment:
Oh, wait. Nevermind. I think this is okay. The package gets "added to the result set" , even though it is updated (not inserted) in the collection.